File: Objective-C/UIKit/UIButton/UIButton+Image.m
//
//  UIButton+Image.m
//  UIButtonTest
//
//  Created by 巴糖 on 2017/8/23.
//  Copyright © 2019年 crazyit.org. All rights reserved.
//

#import "UIButton+Image.h"

@implementation UIButton (Image)

- (UIImage *)imageForNormal{
    return [self imageForState:UIControlStateNormal];
}

- (UIImage *)imageForHighlighted{
    return [self imageForState:UIControlStateHighlighted];
}

- (UIImage *)imageForDisabled{
    return [self imageForState:UIControlStateDisabled];
}

- (UIImage *)imageForSelected{
    return [self imageForState:UIControlStateSelected];
}


- (void)setImageForNormal:(nullable UIImage *)image{
    [self setImage:image forState:UIControlStateNormal];
}

- (void)setImageForHighlighted:(nullable UIImage *)image{
    [self setImage:image forState:UIControlStateHighlighted];
}

- (void)setImageForDisabled:(nullable UIImage *)image{
    [self setImage:image forState:UIControlStateDisabled];
}

- (void)setImageForSelected:(nullable UIImage *)image{
    [self setImage:image forState:UIControlStateSelected];
}

@end

File: Swift/UIKit/UIButton+Image.swift
//
//  UIButton+Image.swift
//  vshiMessge
//
//  Created by 巴糖 on 2017/7/28.
//  Copyright © 2017年 巴糖. All rights reserved.
//

import UIKit

// MARK: - Properties
public extension UIButton {

    @IBInspectable
    /// Image of normal state for button; also inspectable from Storyboard.
    public var imageForNormal: UIImage? {
        get {
            return image(for: .normal)
        }
        set {
            setImage(newValue, for: .normal)
        }
    }
    
    @IBInspectable
    /// Image of highlighted state for button; also inspectable from Storyboard.
    public var imageForHighlighted: UIImage? {
        get {
            return image(for: .highlighted)
        }
        set {
            setImage(newValue, for: .highlighted)
        }
    }
    
    @IBInspectable
    /// Image of selected state for button; also inspectable from Storyboard.
    public var imageForSelected: UIImage? {
        get {
            return image(for: .selected)
        }
        set {
            setImage(newValue, for: .selected)
        }
    }
    
    @IBInspectable
    /// Image of disabled state for button; also inspectable from Storyboard.
    public var imageForDisabled: UIImage? {
        get {
            return image(for: .disabled)
        }
        set {
            setImage(newValue, for: .disabled)
        }
    }
}
